File: day20/particle_swarm.pl
#!/usr/bin/env perl
use strict;
use warnings;
use feature qw(:5.24);
use experimental qw(signatures);
use lib '.';
use Particle;

my @part;

while (<>) {
    chomp;
    push @part, Particle->new($_);
}

my $low_accel = 1e300;
my @candidates;

for my $i (0..$#part) {
    my $mag = $part[$i]->accel_mag();
    if ($mag < $low_accel) {
        $low_accel = $mag;
        @candidates = ($i);
    } elsif ($mag == $low_accel) {
        push @candidates, $i;
    }
}
say "result1: @candidates";

# if there were a tie, we'd also need to compare initial velocities

File: day20/particle_swarm.py
#!/usr/bin/env python3
from sys import argv
from Particle import *

part=[]
filename = argv[1]
with open(filename) as f:
    for line in f:
        part.append(Particle(line.rstrip()))

# for i in range(len(part)):
#     print(i, part[i].dump())

low_accel = 1e300
candidates = []

for i in range(len(part)):
    mag = part[i].accel_mag()
    if mag < low_accel:
        low_accel = mag
        candidates = [i]
    elif mag == low_accel:
        candidates.append(i)

print("result1:", candidates)

# if there were a tie, we'd also need to compare initial velocities
